src/Application/Actions/Donations/Create.php

```diff
@@ -40,9 +41,16 @@ public function __construct(
     /**
      * @return Response
      * @throws \MatchBot\Application\Matching\TerminalLockException if the matching adapter can't allocate funds
+     * @see PersonManagementAuthMiddleware
      */
     protected function action(): Response
     {
+        // The route at `/people/{personId}/donations` validates that the donor has permission to act
+        // as the person, and sets this attribute to the Stripe Customer ID based on JWS claims, all
+        // in `PersonManagementAuthMiddleware`. If the legacy route was used or if no such ID was in the
+        // JWS, this is null.
+        $customerId = $this->request->getAttribute('pspId');
```

Comment:
@NoelLH I might be mistaken but shouldn't this be `$this->request->getAttribute('personId');`?

Reply:
@alihejazi Nope, don't think so. Check out the `$request->withAttribute(...)` call and logic leading up to it in the middleware.